Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+
+Binary packages for macOS are stored as `.tgz`, whereas Windows binary packages end in `.zip`. If you need to make a binary package, use `devtools::build(binary = TRUE)` on the relevant operating system. Under the hood, this calls `pkgbuild::build(binary = TRUE)` and, ultimately, `R CMD INSTALL --build`, which is described further in the [Building binary packages][wre-binary] secion of [Writing R Extensions][wre].
+
+CRAN packages are usually available in binary form, for both macOS and Windows, for the current, previous, and (possibly) development versions of R. Continuing our examples from above, you could download binary packages such as:
+
+  * forcats for macOS: `forcats_0.4.0.tgz`
+  * stringr for Windows: `stringr_1.4.0.zip`
+
+and this is, indeed, part of what's usually going on behind the scenes when you call `install.packages()`.
+
+If you uncompress a binary package, you'll see that the internal structure is rather different from a source or bundled package. Here are some of the most notable differences:
+
+* There are no `.R` files in the `R/` directory - instead there are three
+  files that store the parsed functions in an efficient file format. This is
+  basically the result of loading all the R code and then saving the
+  functions with `save()`. (In the process, this adds a little extra metadata to 
+  make things as fast as possible).
+
+* A `Meta/` directory contains a number of `.rds` files. These files contain 
+  cached metadata about the package, like what topics the help files cover and
+  a parsed version of the `DESCRIPTION` file. (You can use `readRDS()` to see 
+  exactly what's in those files). These files make package loading faster
+  by caching costly computations.
+
+* The actual help content appears in `help/` and `html/` (no longer in `man/`).
+
+* If you had any code in the `src/` directory, there will now be a `libs/`
+  directory that contains the results of compiling the code. On Windows, there
+  are subdirectories for 32 bit (`i386/`) and 64 bit (`x64/`) environments.
+
+* If you had any objects in `data/`, they have now been converted into a more
+  efficient form.
+  
+* The contents of `inst/` are moved to the top-level directory. For example,
+  vignette files are now in `doc/`.
+
+* Some files and folders have been dropped, such as `README`, `build/`, `tests/`,
+  and `vignettes/`.
+
+### Installed package
+
+An __installed__ package is a binary package that's been decompressed into a package library (described in \@ref(library)). The following diagram illustrates the many ways a package can be installed. This diagram is complicated! In an ideal world, installing a package would involve stringing together a set of simple steps: source -> bundle, bundle -> binary, binary -> installed. In the real world, it's not this simple because there are often (faster) shortcuts available.
+
+```{r, echo = FALSE}
```

Comment:
Figures usually should have `output.width = NULL` so that they're sized naturally according to their actual size. For large figures (like this one?), you should probably also add a caption so that they float.

Reply:
I think the relevant chunk option is `out.width` (???) and I couldn't get happy things to happen with `NULL` but I did with `out.width = "100%"`. I imagine my understanding of this and general figure mechanics are sub-par but will be forced to improve.

Added captions to two figures covered by this PR and will continue adding as I work through the book.